File: archieve/frontend_react/src/types/finance.ts
export interface FinancialCategory {
  id: number;
  user_id: number;
  name: string;
  type: 'income' | 'expense';
}

export interface FinancialTransaction {
  id: number;
  user_id: number;
  category_id?: number;
  amount: number;
  description?: string;
  date: string;
  type: 'income' | 'expense';
}

export type FinancialCategoryCreate = Omit<FinancialCategory, 'id' | 'user_id'>;
export type FinancialCategoryUpdate = Partial<FinancialCategoryCreate>;

export type FinancialTransactionCreate = Omit<FinancialTransaction, 'id' | 'user_id'>;
export type FinancialTransactionUpdate = Partial<FinancialTransactionCreate>;

File: archieve/backend_python/app/schemas/finance.py
from pydantic import BaseModel
from datetime import date
from enum import Enum

class TransactionType(str, Enum):
    INCOME = "income"
    EXPENSE = "expense"

class FinancialCategoryBase(BaseModel):
    name: str
    type: TransactionType

class FinancialCategoryCreate(FinancialCategoryBase):
    pass

class FinancialCategoryUpdate(FinancialCategoryBase):
    pass

class FinancialCategory(FinancialCategoryBase):
    id: int
    user_id: int

    class Config:
        orm_mode = True

class FinancialTransactionBase(BaseModel):
    amount: float
    description: str | None = None
    date: date
    type: TransactionType
    category_id: int | None = None

class FinancialTransactionCreate(FinancialTransactionBase):
    pass

class FinancialTransactionUpdate(FinancialTransactionBase):
    pass

class FinancialTransaction(FinancialTransactionBase):
    id: int
    user_id: int

    class Config:
        orm_mode = True
